### PR TITLE
RW-11041 update sas, rstudio chart reference

### PR DESCRIPTION
**Pull request by Qi77Qi**
Related PRs:
https://github.com/all-of-us/workbench/pull/8196
https://github.com/broadinstitute/terra-helmfile/pull/4796
https://github.com/broadinstitute/terra-helmfile/pull/4795

Just like other data plane charts, we're moving sas and rstudio charts to terra-helmfile repo as well. Hence we need to update the chart references in a few places. @jack-r-warren mentioned it's better to not have `chart` in the chart name, so chart names are renamed as well.

This also bumps SAS charts for a bug fix.

Tested in BEE for creating both RStudio and SAS app

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala [hunk 1]
```diff
@@ -316,14 +316,22 @@ sealed abstract class AllowedChartName extends Product with Serializable {
...
   }
   final case object Sas extends AllowedChartName {
-    def asString: String = "aou-sas-chart"
+    def asString: String = "sas"
     def trackUsage: Boolean = true
   }
-  def stringToObject: Map[String, AllowedChartName] = sealerate.values[AllowedChartName].map(v => v.asString -> v).toMap
+
+  // We used to have different chart names for RStudio and SAS. This is to handle the old names for backwards-compatibility
+  private val deprecatedName: Map[String, AllowedChartName] = Map(
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala:328]: this serves 2 purposes
1. for backwards-compatibility at contract level. Meaning old value will be accepted
2. for backwards-compatibility at DB level, meaning `getApp` will function properly for old apps that have old chart names

**Review by yonghaoy — APPROVED** [review]
Sorry, I missed the previous rename chart PR. About dropping `aou` from chart name: 
those charts are actually for aou, e.g.. they have Wondershaper installed

**Comment by Qi77Qi** [comment]
@yonghaoy yea good point..but I think we can add a flag for whether wondershaper is enabled once we need to use the charts for Terra tho...we don't need to duplicate the entire charts

**Review by LizBaldo — APPROVED** [review]

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
```
17:03:30 logs: https://console.cloud.google.com/storage/browser/thelma-artifacts-terra-qa-bees/jenkins-swat-leo-12525
17:03:36 22:03 ERR error="1 execution errors:\nmartha: timed out waiting for healthy (Degraded: martha-deployment: Back-off restarting failed container)\n"
17:03:36 Build step 'Conditional steps (multiple)' marked build as failure
```

**Pull request merged**